extension/demo/src/bg/background.js

```diff
@@ -23,44 +24,59 @@ function onPopupOpenApps () {
 
 async function onToken (data) {
   const { accessToken, subscription } = data
-  let payload = {}
-  payload[STORAGE_KEY] = {
-    auth: {
-      accessToken: accessToken,
-      subscription: subscription
-    }
-  }
-  await chrome.storage.local.set(payload)
-  const isAuthenticated = !chrome.runtime.lastError
+  const isAuthenticated = await storage.set({ accessToken, subscription })
   sendMessage('popup-set-view', {
     isAuthenticated
   })
+
   return {
     isAuthenticated
   }
 }
 
+function timeToLive (expiresAt) {
+  return expiresAt - (Math.floor(Date.now() / 1000) + 3600)
+}
+
+async function updatePayload () {
+  const payload = await storage.get(PAYLOAD)
+  if (!payload) {
+    const response = await api.fetchPayload()
+    if (response.hasOwnProperty(PAYLOAD)) { // fetch return might need to go through .json()?
+      let { expiresAt, payload } = response
+      await storage.set({ expiresAt, payload })
+    }
+  } else {
+    const ttl = await storage.get(TTL)
+    if (ttl) {
+      setTimeout(updatePayload, timeToLive(ttl))
+    }
+  }
+}
+
 async function onSubscription (data) {
   const { subscription } = data
-  let payload = {}
-  payload[STORAGE_KEY]['auth']['subscription'] = subscription
-  await chrome.storage.local.set(payload)
-  chrome.storage.local.get([STORAGE_KEY], results => {
-    console.log(results)
-    return results
-  })
+  await storage.set({ subscription })
 }
```

Comment:
No need to use `await` here as there's nothing we need to wait for. If you want to return a promise I would instead remove the `async/await` and just return the `storage.set` function